sumpy/expansion/local.py

```diff
@@ -47,6 +43,31 @@ class LocalExpansionBase(ExpansionBase):
 """
 
 
+class LocalExpansionBase(ExpansionBase):
+    """Base class for local expansions.
+    """
+    def m2l_global_precompute_exprs(self, src_expansion, src_rscale,
+            dvec, tgt_rscale, sac):
+        """Return an iterable of expressions that needs to be precomputed
+        for multipole-to-local translations that depend only on the
+        distance between the multipole center and the local center.
+
+        Since there are only a finite amount of different values for the
+        distance between per level, these can be precomputed for the tree.
+        """
+        return tuple()
+
+    def m2l_global_precompute_nexpr(self, src_expansion):
+        """Return the number of expressions returned by
+        :func:`~sumpy.expansion.local.LocalExpansionBase.m2l_global_precompute_exprs`.
+        """
+        return 0
+
+    def translate_from(self, src_expansion, src_coeff_exprs, src_rscale,
+            dvec, tgt_rscale, sac=None, precomputed_exprs=None):
```

Comment:
What's the type of `precomputed_exprs`? What assumptions are made about it? Can I pass anything and it'll be OK? Or is it supposed to be exactly what came out of `m2l_global_precompute_exprs`?